src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -615,39 +637,38 @@ private boolean verifyAuthenticator(final AccountManager am) {
      */
     @Override
     public UserInfo[] getBrokerUsers() throws OperationCanceledException, AuthenticatorException, IOException {
-
+        final String methodName = ":getBrokerUsers";
         // Calling this on main thread will cause exception since this is
         // waiting on AccountManagerFuture
         if (Looper.myLooper() == Looper.getMainLooper()) {
             throw new IllegalArgumentException("Calling getBrokerUsers on main thread");
         }
 
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        Bundle bundle = new Bundle();
+        final Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
+        final Bundle bundle = new Bundle();
         bundle.putBoolean(DATA_USER_INFO, true);
-
-        if (accountList != null) {
-
-            // get info for each user
-            UserInfo[] users = new UserInfo[accountList.length];
-            for (int i = 0; i < accountList.length; i++) {
-
-                // Use AccountManager Api method to get extended user info
-                AccountManagerFuture<Bundle> result = mAcctManager.updateCredentials(accountList[i],
-                        AuthenticationConstants.Broker.AUTHTOKEN_TYPE, bundle, null, null, null);
-                Logger.v(TAG, "Waiting for the result");
-                Bundle userInfoBundle = result.getResult();
-
-                users[i] = new UserInfo(
-                        userInfoBundle.getString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_USERID),
-                        userInfoBundle.getString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_GIVEN_NAME),
-                        userInfoBundle.getString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_FAMILY_NAME),
-                        userInfoBundle.getString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_IDENTITY_PROVIDER),
-                        userInfoBundle.getString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_USERID_DISPLAYABLE));
-            }
-
-            return users;
+        Logger.v(TAG + methodName, "Retrieve all the accounts from account manager with broker account type, "
+                + "and the account length is: " + accountList.length);
+
+        // accountList will never be null, getAccountsByType will return an empty list if no matching account returned.
+        // get info for each user
+        final UserInfo[] users = new UserInfo[accountList.length];
```

Comment:
We should definitely document the behavior. But changing the possible null return to non-null array won't breaking any client side existing behavior. The opposite way will do.